cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -4144,31 +4152,34 @@ setupHsConfigureFlags
       configHcFlavor = toFlag (compilerFlavor pkgConfigCompiler)
       configHcPath = mempty -- we use configProgramPaths instead
       configHcPkg = mempty -- we use configProgramPaths instead
-      configVanillaLib = toFlag elabVanillaLib
-      configSharedLib = toFlag elabSharedLib
-      configStaticLib = toFlag elabStaticLib
+      configVanillaLib = toFlag $ LBC.withVanillaLib elabBuildOptions
```

Comment:
Question `BuildOptions` is a Cabal concept and so is `ConfigFlags`. I think it's reasonable to have a `BuildOptions -> ConfigFlags` function in Cabal itself (with `mempty` for everything missing).

After all how to get `Cabal` talk to itself should not be `cabal-install`'s responsibility.

Reply:
Good idea, I added a `buildOptionsConfigFlags` function in `Cabal` and used that here in `cabal-install`.